src/write_outputs/write_vre_stor.jl

```diff
@@ -0,0 +1,408 @@
+@doc raw"""
+	write_vre_stor(path::AbstractString, inputs::Dict, setup::Dict, EP::Model)
+
+Function for writing the vre-storage specific files.
+"""
+
+function write_vre_stor(path::AbstractString, inputs::Dict, setup::Dict, EP::Model)
+
+	### CAPACITY DECISIONS ###
+	dfVreStor = write_vre_stor_capacity(path, inputs, setup, EP)
+
+	### CHARGING DECISIONS ###
+	write_vre_stor_charge(path, inputs, setup, EP)
+
+	### DISCHARGING DECISIONS ###
+	write_vre_stor_discharge(path, inputs, setup, EP)
+
+	return dfVreStor
+end
+
+@doc raw"""
+	write_vre_stor_capacity(path::AbstractString, inputs::Dict, setup::Dict, EP::Model)
+
+Function for writing the vre-storage capacities.
+"""
+function write_vre_stor_capacity(path::AbstractString, inputs::Dict, setup::Dict, EP::Model)
+	VRE_STOR = inputs["VRE_STOR"]
+	SOLAR = inputs["VS_SOLAR"]
+	WIND = inputs["VS_WIND"]
+	DC = inputs["VS_DC"]
+	STOR = inputs["VS_STOR"]
+	dfGen = inputs["dfGen"]
+	dfVRE_STOR = inputs["dfVRE_STOR"]
+	MultiStage = setup["MultiStage"]
+	size_vrestor_resources = size(inputs["RESOURCES_VRE_STOR"])
+
+	# Solar capacity
+	capsolar = zeros(size_vrestor_resources)
+	retcapsolar = zeros(size_vrestor_resources)
+	existingcapsolar = zeros(size_vrestor_resources)
+
+	# Wind capacity
+	capwind = zeros(size_vrestor_resources)
+	retcapwind = zeros(size_vrestor_resources)
+	existingcapwind = zeros(size_vrestor_resources)
+
+	# Inverter capacity
+	capdc = zeros(size_vrestor_resources)
+	retcapdc = zeros(size_vrestor_resources)
+	existingcapdc = zeros(size_vrestor_resources)
+
+	# Grid connection capacity
+	capgrid = zeros(size_vrestor_resources)
+	retcapgrid = zeros(size_vrestor_resources)
+	existingcapgrid = zeros(size_vrestor_resources)
+
+	# Energy storage capacity
+	capenergy = zeros(size_vrestor_resources)
+	retcapenergy = zeros(size_vrestor_resources)
+	existingcapenergy = zeros(size_vrestor_resources)
+
+	# Charge storage capacity DC
+	capchargedc = zeros(size_vrestor_resources)
+	retcapchargedc = zeros(size_vrestor_resources)
+	existingcapchargedc = zeros(size_vrestor_resources)
+
+	# Charge storage capacity AC
+	capchargeac = zeros(size_vrestor_resources)
+	retcapchargeac = zeros(size_vrestor_resources)
+	existingcapchargeac = zeros(size_vrestor_resources)
+
+	# Discharge storage capacity DC
+	capdischargedc = zeros(size_vrestor_resources)
+	retcapdischargedc = zeros(size_vrestor_resources)
+	existingcapdischargedc = zeros(size_vrestor_resources)
+
+	# Discharge storage capacity AC
+	capdischargeac = zeros(size_vrestor_resources)
+	retcapdischargeac = zeros(size_vrestor_resources)
+	existingcapdischargeac = zeros(size_vrestor_resources)
+	
+	j = 1
+	for i in VRE_STOR
+		existingcapgrid[j] = MultiStage == 1 ? value(EP[:vEXISTINGCAP][i]) : dfGen[!,:Existing_Cap_MW][i]
```

Comment:
`dfGen[!,:Existing_Cap_MW][i]` --> `dfGen[i,:Existing_Cap_MW]` for all these. Careful of `i` vs `j`.